libs/prelude/Builtins.idr

```diff
@@ -213,3 +213,8 @@ export data ManagedPtr : Type
 
 %extern prim__peekPtr : prim__WorldType -> Ptr -> Int -> Ptr
 %extern prim__pokePtr : prim__WorldType -> Ptr -> Int -> Ptr -> Int
+
+%extern prim__peekDouble : prim__WorldType -> Ptr -> Int -> Double
+%extern prim__pokeDouble : prim__WorldType -> Ptr -> Int -> Double -> Int
+%extern prim__peekSingle : prim__WorldType -> Ptr -> Int -> Double
+%extern prim__pokeSingle : prim__WorldType -> Ptr -> Int -> Double -> Int
```

Comment:
They are primitives because I believe that a general mechanism for accessing something memory-like would be useful for most possible backends, even Javascript has the tools to implement them these days. I thought it was reasonable to handle floats the same as integer types.

The cop-out answer for how to implement them is that one doesn't have to. They're external primitives, they're not mandatory (not that internal primitives are mandatory either :-)). But I made the interface Ptr + Offset is that Ptr could represent for example, a memoryview object in Python or a JavaScript ArrayBuffer..